### ISSUE TITLE
glibc-2.31 & hwclock

### ISSUE THREAD

**Issue by udeved** [OpenRC/openrc#352]
Starting with glibc-2.31, the hwclock throws an error on boot.

` 
* Setting system clock using the hardware clock [UTC] ...
hwclock: settimeofday() failed: Invalid argument
 * Failed to set the system clock
 [ !! ]
`

**Comment by udeved** [comment]
glibc changelog ... https://sourceware.org/git/?p=glibc.git;a=blob_plain;f=NEWS;hb=HEAD

**Comment by trofi** [comment]
Right, relevant entry is:
```
* The settimeofday function can still be used to set a system-wide time
  zone when the operating system supports it.  This is because the Linux
  kernel reused the API, on some architectures, to describe a system-wide
  time-zone-like offset between the software clock maintained by the kernel,
  and the "RTC" clock that keeps time when the system is shut down.

  However, to reduce the odds of this offset being set by accident,
  settimeofday can no longer be used to set the time and the offset
  simultaneously.  If both of its two arguments are non-null, the call
  will fail (setting errno to EINVAL).

  Callers attempting to set this offset should also be prepared for the call
  to fail and set errno to ENOSYS; this already happens on the Hurd and on
  some Linux architectures.  The Linux kernel maintainers are discussing a
  more principled replacement for the reused API.  After a replacement
  becomes available, we will change settimeofday to fail with ENOSYS on all
  platforms when its 'tzp' argument is not a null pointer.

  settimeofday itself is obsolescent according to POSIX.  Programs that set
  the system time should use clock_settime and/or the adjtime family of
  functions instead.  We may cease to make settimeofday available to newly
  linked binaries after there is a replacement for Linux's time-zone-like
  offset API.
```
and probably comes from `/sbin/hwclock` (sys-apps/util-linux package).

sys-apps/util-linux was fixed upstream in https://github.com/karelzak/util-linux/commit/cd781c405be82540484da3bfe3d3f17a39b8eb5c. Don't know if Gentoo backported it.

**Comment by williamh** [comment]
This is fixed in util-linux 2.36.x.
